hikari/impl/buckets.py

```diff
@@ -506,50 +557,32 @@ def acquire(self, compiled_route: routes.CompiledRoute) -> asyncio.Future[None]:
 
         Returns
         -------
-        asyncio.Future[builtins.None]
-            A future to await that completes when you are allowed to run
-            your request logic.
+        typing.AsyncContextManager[builtins.None]
+            A context manager to enter while doing the request.
 
         !!! note
-            The returned future MUST be awaited, and will complete when your
-            turn to make a call comes along. You are expected to await this and
-            then immediately make your HTTP call. The returned future may
-            already be completed if you can make the call immediately.
+            You MUST keep the context manager acquired during the whole of the
+            request. From making the request until calling `update_rate_limits`.
         """
-        # Returns a future to await on to wait to be allowed to send the request, and a
-        # bucket hash to use to update rate limits later.
         template = compiled_route.route
 
-        if template in self.routes_to_hashes:
+        try:
             bucket_hash = self.routes_to_hashes[template]
-        else:
+            real_bucket_hash = compiled_route.create_real_bucket_hash(bucket_hash)
+        except KeyError:
             bucket_hash = UNKNOWN_HASH
             self.routes_to_hashes[template] = bucket_hash
```

Comment:
Shouldn't the generated pseudo hash be used here not "UNKNOWN" since that'd never resolve to a valid bucket?